src/components/Charts/LineChart.tsx

```diff
@@ -28,80 +38,208 @@ import LineChartOperations from '@/operations/charts/LineChartOperations';
  */
-export function LineChart({ title, description, expandable, small, data }: LineChartProps) {
+export function LineChart({
+  title,
+  description,
+  expandable,
+  small,
+  barChartSwitch,
+  xAxisSlider,
+  transparentBackground,
+  data,
+}: LineChartProps) {
   const TITLE_TEXT_SIZE = small ? 'text-sm' : 'text-md';
   const DESCRIPTION_TEXT_SIZE = small ? 'text-xs' : 'text-sm';
   const CHART_HEIGHT = small ? 12 : 16;
   const ICON_BUTTON_SIZE = small ? 3 : 4;
-  const HEADER_BOTTOM_PADDING = title ? 3 : 0;
-  const JSON_DOWNLOAD_FILE_NAME = `hunger_map_line_chart_json-${title}.json`;
+  const HEADER_PADDING = title ? 3 : 0;
 
+  // referencing the Highcharts chart object (needed for download the chart as a png)
+  const chartRef = useRef<HighchartsReact.RefObject | null>(null);
   // full screen modal state handling
-  const { isOpen, onOpen, onOpenChange } = useDisclosure();
+  const { isOpen, onOpen, onClose, onOpenChange } = useDisclosure();
+  // the 'chartOptions' are dependent on the theme
+  const { theme } = useTheme();
 
-  // convert data to `LineChartData` and build chart options for 'Highcharts'
+  // convert data to `LineChartData` and build chart options for 'Highcharts' (line and bar chart)
   const lineChartData: LineChartData = LineChartOperations.convertToLineChartData(data);
-  const chartOptions: Highcharts.Options = LineChartOperations.getHighChartData(lineChartData);
+  const lineChartOptions: Highcharts.Options = LineChartOperations.getHighChartLineData(lineChartData, theme);
+
+  // controlling if a line or bar chart is rendered; line chart is the default
+  const [showBarChart, setShowBarChart] = useState(false);
+  const [chartOptions, setChartOptions] = useState(lineChartOptions);
+  // handling the x-axis range slider visibility
+  const [showXAxisSlider, setShowXAxisSlider] = useState(false);
 
-  // trigger download of the given line chart `data` as a json file
-  const downloadDataJson = () => {
-    // convert data json object to string and encode as URI
-    const jsonString = `data:text/json;charset=utf-8,${encodeURIComponent(JSON.stringify(data, null, 2))}`;
-    // create a temporary link element and trigger the download
-    const link = document.createElement('a');
-    link.href = jsonString;
-    link.download = JSON_DOWNLOAD_FILE_NAME;
-    link.click();
-  };
+  // handling the line and bar chart switch and the theme switch
+  useEffect(() => {
+    const newChartOptions = showBarChart
+      ? LineChartOperations.getHighChartBarData(lineChartData, theme)
+      : LineChartOperations.getHighChartLineData(lineChartData, theme);
+    setChartOptions(newChartOptions);
+  }, [showBarChart, theme]);
 
-  // Full screen modal that can be opened if `expandable==true`. Offers a larger chart and a download button.
+  // full screen modal that can be opened if `expandable==true`; offers a larger chart and an additional features (see buttons)
   const fullScreenModal = (
```

Comment:
to answer your question this could be a separate component and you could pass the state and callbacks via params as usual